docs/design/cronjob.md

```diff
@@ -0,0 +1,147 @@
+# CronJob
+
+[@TommyLike](https://github.com/tommyLike/); April 23, 2019
+
+## Motivation
+
+`CronJob` is designed to help end user to schedule their batch job at some specific time, such as if you want to
+schedule a batch job at a low activity period, or just need the job to be executed periodically, it's especially
```

Comment:
I can see the value of delay scheduling/running, but can hardly imagine when  periodically run needed. Can you elaborate a little?

Reply:
It's a common case for high performance tasks which usually task a long time, for example, daily processing business data, generate report, update warehouse.

Reply:
CronJob is valueable, add the flexibility of scheduler.

Reply:
yes, we need this feature :)